business/namespaces.go

```diff
@@ -493,26 +584,23 @@ func (in *NamespaceService) getNamespacesUsingKialiSA(labelSelector string, forw
 	return namespaces, nil
 }
 
-func getNamespacesForKialiSA(labelSelector string) ([]core_v1.Namespace, error) {
-	clientFactory, err := kubernetes.GetClientFactory()
-	if err != nil {
-		return nil, err
-	}
-
-	kialiToken, err := kubernetes.GetKialiTokenForHomeCluster()
-	if err != nil {
-		return nil, err
-	}
+func (in *NamespaceService) getNamespacesForKialiSA(cluster string, labelSelector string) ([]core_v1.Namespace, error) {
```

Comment:
After looking more into what's going on inside `GetNamespaces`, it looks like the kiali service account token is being used when `accessible_namespaces == **`. Kiali tries to do a `GetNamespaces` with the user client: https://github.com/kiali/kiali/blob/f3f431063e067dd58c0906008e9a4c915e75a6e4/business/namespaces.go#L182-L192.

If that fails, then Kiali does a `GetNamespaces` with the Kiali Service Account client to get the whole list of namespaces and then does `GetNamespace(...)` call against each individually to check the user's access to each one.
https://github.com/kiali/kiali/blob/f3f431063e067dd58c0906008e9a4c915e75a6e4/business/namespaces.go#L474-L490

When `accessible_namespaces == **` _and_ the logged in user doesn't have `List Namespaces` access to a cluster, the Namespace Service uses the Kiali service account client to retrieve the list of namespaces and check the user's access to each one. So the Namespace Service needs the Kiali service account client for this. IMO it's still best to pass the kiali service account clients into the namespace service rather than using the client factory from inside the Namespace Service. That'd look something like:
```
type NamespaceService struct {
  userClients map[string]ClientInterface
  kialiSAClients map[string]ClientInterface
}
```
that might require a change in `NewWithBackends`. Either passing in the clientfactory rather than `map[string]ClientInterface` or passing in two different client maps, one for the user clients and one for the sa clients.

Reply:
Yes, change addressed. Also, code looks much more clear now.